taskfiles/utils.yml

```
@@ -1,64 +1,87 @@
 version: "3"
 
+set: ["u", "pipefail"]
+shopt: ["globstar"]
+
 tasks:
   # ===
   # CHECKSUM UTILS
   # ===
 
-  # @param {string} DATA_DIR The directory to compute the checksum for.
+  # @param {string[]} INCLUDE_PATTERNS Path wildcard patterns to compute the checksum for.
   # @param {string} OUTPUT_FILE
-  # @param {[]string} [EXCLUDE_PATHS] A list of paths, relative to `DATA_DIR`, to exclude from the
-  # checksum.
+  # @param {string[]} [EXCLUDE_PATTERNS] Path wildcard patterns, relative to any `INCLUDE_PATTERNS`,
+  # to exclude from the checksum.
   compute-checksum:
-    desc: "Tries to compute a checksum for the given directory and output it to a file."
+    desc: "Tries to compute a checksum for the given paths and output it to a file."
     internal: true
+    label: "{{.TASK}}-{{.OUTPUT_FILE}}"
     silent: true
     requires:
-      vars: ["DATA_DIR", "OUTPUT_FILE"]
+      vars: ["INCLUDE_PATTERNS", "OUTPUT_FILE"]
     cmds:
+      # We explicitly set --no-anchored and --wildcards to ensure the behaviour for both exclusion
+      # and inclusion matches, as they are only the default for exclusion.
+      # We can quote the exclude patterns for tar because the patterns themselves are passed to tar
+      # as arguments and evaluated by tar during execution. However, for tar --create the input
+      # patterns cannot be quoted as they are evaluated by the shell and the results are passed to
+      # tar as arguments. If the input patterns are passed to tar with quotes the pattern will not
+      # be evaluated and will be treated literally.
       - >-
-        tar cf -
-        --directory "{{.DATA_DIR}}"
+        tar
+        --create
+        --file -
         --group 0
         --mtime "UTC 1970-01-01"
         --numeric-owner
         --owner 0
         --sort name
-        {{- range .EXCLUDE_PATHS}}
+        --no-anchored
+        --wildcards
+        {{- range .EXCLUDE_PATTERNS}}
         --exclude="{{.}}"
         {{- end}}
-        . 2> /dev/null
+        {{- range .INCLUDE_PATTERNS}}
+        {{.}}
+        {{- end}}
+        2> /dev/null
         | md5sum > {{.OUTPUT_FILE}}
     # Ignore errors so that dependent tasks don't fail
     ignore_error: true
 
-  # @param {string} DATA_DIR The directory to validate the checksum for.
+  # @param {string[]} INCLUDE_PATTERNS Path wildcard patterns to validate the checksum for.
   # @param {string} OUTPUT_FILE
```

```suggestion
  # @param {string} CHECKSUM_FILE
  # @param {string[]} INCLUDE_PATTERNS Path wildcard patterns to validate the checksum for.
```

Note the rename from OUTPUT_FILE to CHECKSUM_FILE.